### PR TITLE
feat(applications-service-api): complete submission endpoint

### PR DESCRIPTION
**Pull request by JamesP-KC**
- new `/submissions/{submissionId}/complete` endpoint
  - updated openapi spec
  - stub implementation
      - request validation
      - fetching submission from DB
      - does not call notify service yet, to be implemented in ASB-901
 - fix incorrect path of create submission endpoint. `/submission/:caseReference` -> `/submissions/:caseReference`

### CODE REVIEW

File: packages/applications-service-api/src/error/apiError.js [hunk 1]
```diff
@@ -21,39 +25,27 @@ class ApiError {
...
-			errors: [`Interested party for project ${caseRef} not found`]
-		});
+		return this.notFound(`Interested party for project ${caseRef} not found`);
 	}
 
 	static noDocumentsFound() {
-		return new ApiError(404, { errors: [`No documents found`] });
-	}
-
-	static commentsForPartyWithIDNotUpdated(ID) {
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/error/apiError.js:37]: Is this a not found or could something else result in this error?

File: packages/applications-service-api/src/error/apiError.js [hunk 2]
```diff
@@ -10,8 +10,12 @@ class ApiError {
 		return new ApiError(StatusCodes.BAD_REQUEST, { errors: messages.flat() });
 	}
 
+	static notFound(message) {
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/error/apiError.js:13]: Nice tidy up of the error code.

File: packages/applications-service-api/src/services/submission.service.js [hunk 3]
```diff
@@ -66,7 +67,22 @@ const updateSubmission = async (submission) => {
...
+const getSubmission = (submissionId) => db.Submission.findOne({ where: { id: submissionId } });
+
+const completeSubmission = async (submissionId) => {
+	const submission = await getSubmission(submissionId);
+
+	if (!submission) {
+		throw ApiError.notFound(`Submission with ID ${submissionId} not found`);
+	}
+
+	console.log(`Found submission with ID ${submission.id}, email: ${submission.email}`);
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/services/submission.service.js:79]: I don't think we should log the email as it's personal information.

File: packages/applications-service-api/src/error/apiError.js (hunk 1 above)
Comment (JamesP-KC) [reply packages/applications-service-api/src/error/apiError.js:37]: I removed this `commentsForPartyWithIDNotUpdated` function as it doesn't seem to be in use anywhere

File: packages/applications-service-api/src/services/submission.service.js (hunk 3 above)
Comment (JamesP-KC) [reply packages/applications-service-api/src/services/submission.service.js]: That's fair enough, I added this line just temporarily so the endpoint does "something" and was going to remove it in the next story. It's not really needed so I will remove now

**Review by Philip-Philippou1 — APPROVED** [review]
Looks good

**Pull request merged**